src/lib.rs

```diff
@@ -0,0 +1,28 @@
+use std::net::SocketAddr;
+use clap::Parser;
+use hyper::Error;
+use routing::{get_main_router};
+use tracing::subscriber::SetGlobalDefaultError;
+use tracing_subscriber::FmtSubscriber;
+
+// declare child modules
+mod routing;
+
+#[derive(Debug, Parser)]
+pub struct LaunchConfig {
+    pub addr: SocketAddr
+}
```

Comment:
I don't see why addr have to set on run. Or do you plan to load configs from separate file?

Reply:
Hard coding constants is bad practice

Reply:
Even though, It might be nice to have the ports written in some config. Or at least set some defaults.

Reply:
Maybe a .env or .ini file? I don't like the idea of the defaults being in the code.

Reply:
I just made a rust file with some constants for now, we'll see if there's more things that could go into a config file in the future and then maybe change it to use some sort of configuration file.